control/plugin/cpolicy/bool.go

```diff
@@ -0,0 +1,152 @@
+/*
+http://www.apache.org/licenses/LICENSE-2.0.txt
+
+
+Copyright 2016 Intel Corporation
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package cpolicy
+
+import (
+	"bytes"
+	"encoding/gob"
+	"encoding/json"
+
+	"github.com/intelsdi-x/snap/core/ctypes"
+)
+
+// BoolRule validating against string-typed config
+type BoolRule struct {
+	rule
+
+	key      string
+	required bool
+	default_ *bool
+}
+
+// NewBoolRule Returns a new bool-typed rule. Arguments are key(string), required(bool), default(bool).
+func NewBoolRule(key string, req bool, opts ...bool) (*BoolRule, error) {
+	// Return error if key is empty
+	if key == "" {
+		return nil, EmptyKeyError
+	}
+
+	options := make([]*bool, 1)
```

Comment:
It took me a minute to understand what `options` was for.

You can use <code>opts[_n_]</code>, so maybe after creating your instance below, you could instead do something like this:

```go
b := &BoolRule {
        key:      key,
        required: req,
}

// test whether or not a default was given.
if len(opts) != 0 {
        b.default_ = opts[0]
}
return b, nil
```